feat/feature_test.go

```diff
@@ -288,3 +288,89 @@ func (s *S) TestOrientationWithin(c *check.C) {
 	cycle.loc = &cycle
 	c.Check(func() { feat.OrientationWithin(cycle, chrom1) }, check.Panics, "feat: feature chain too long")
 }
+
+// Tests for BasePositionOf and PositionWithin.
+var baseCoordsTests = []struct {
+	f, ref                  feat.Feature
+	pos, basePos, posWithin int
+}{
+	{
+		f:         chrom1,
+		pos:       20,
+		basePos:   20,
+		posWithin: 20,
+		ref:       chrom1,
+	},
+	{
+		f:         geneA,
+		pos:       30,
+		basePos:   40,
+		posWithin: 40,
+		ref:       chrom1,
+	},
+	{
+		f:         proA,
+		pos:       40,
+		basePos:   60,
+		posWithin: 60,
+		ref:       chrom1,
+	},
+	{
+		f:         opA,
+		pos:       20,
+		basePos:   52,
+		posWithin: 52,
+		ref:       chrom1,
+	},
+	{
+		f:         antiA,
+		pos:       20,
+		basePos:   75,
+		posWithin: 75,
+		ref:       chrom1,
+	},
+	{
+		f:         freeOri,
+		pos:       0,
+		basePos:   10,
+		posWithin: 0,
```

Comment:
Review: Had to add posWithin cause now `PositionWIthin()` has a slightly different meaning that `BasePositionOf()`